modules/dataframe/src/util/dataframeutil.cpp

```diff
@@ -524,6 +524,64 @@ std::shared_ptr<DataFrame> combineDataFrames(std::vector<std::shared_ptr<DataFra
     return newDataFrame;
 }
 
+#include <warn/push>
+#include <warn/ignore/conversion>
+std::vector<std::uint32_t> filteredRows(const Column& col, dataframefilters::Filters filters) {
+
+    if (col.getColumnType() == ColumnType::Categorical) {
+        const auto& catCol = dynamic_cast<const CategoricalColumn&>(col);
+        std::vector<std::uint32_t> rows;
+        for (auto&& [row, value] : util::enumerate<std::uint32_t>(catCol.getValues())) {
+            auto test = util::overloaded{
+                [v = value](const std::function<bool(std::string_view)>& func) { return func(v); },
```

Comment:
since value is a string here, this copy is quite expensive, is there any reason not to do &value or &, nothing should go out of scope here right?

Reply:
value is part of a structured binding not a variable that can be captured. 
> If a lambda-expression [...] captures a structured binding (explicitly or implicitly), the program is ill-formed. 
http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0588r1.html

Reply:
with the transform iterator, `value` is now a `const std::string&`

Reply:
yes, but when you assign it to `v` it will deduce to a `std::string...

Reply:
you need &v = value to avoid the copy.

Reply:
still missing &